### PR TITLE
show notifications for start, stop and delete actions from the tray menu

### PR DESCRIPTION
**Pull request by anjannath**
since we'll be removing the `Error` and `Success` structs, `/stop`, `/delete` would not return a body on success, so we need to make changes in the commander to get this result.

### CODE REVIEW

**Review by gbraad — APPROVED** [review]

**Comment by gbraad** [comment]
Please use a spellchecker for text strings.

File: main.js [hunk 1]
```diff
@@ -186,6 +186,29 @@ createTrayMenu = function(state) {
...
+  response = await commander.start()
+  if (response.KubeletStarted) {
+    showNotification("Cluster started", "CodeReady Containers cluster has started")
+  } else {
+    showNotification("Cluster could not be started", "CodeReady Containers cluster failed to start")
+  }
+}
+
+const handleDeleteClick = async function() {
+  showNotification("Deleteing", "CodeReady Container cluster is being deleted")
```
Comment (gbraad) [inline main.js]: `s/Deleteing/Deleting/`

**Review by gbraad — CHANGES_REQUESTED** [review]
Do we need to add `app.setAppUserModelId("CodeReady Containers");` ?

https://docs.microsoft.com/en-us/windows/win32/shell/appids#how-to-form-an-application-defined-appusermodelid

Perhaps I see the issue because of https://github.com/electron/electron/issues/7934#issuecomment-260078298

**Comment by anjannath** [comment]
> Do we need to add app.setAppUserModelId("CodeReady Containers"); ?

> https://docs.microsoft.com/en-us/windows/win32/shell/appids#how-to-form-an-application-defined-appusermodelid

According to the [section](https://docs.microsoft.com/en-us/windows/win32/shell/appids#how-to-form-an-application-defined-appusermodelid) about ID from the above link it should be `redHat.CodeReadyContainers` or `redHat.CodeReadyContainers.Tray`.

**Comment by anjannath** [comment]
@gbraad The start, stop and delete calls are now using the async, await pattern to handle the errors from the api side. PTALA

**Comment by gbraad** [comment]
> `redHat.CodeReadyContainers.Tray`

Also concluded this. but this needs to be recorded as part of the install to map the appmodel id against a known application name, eg. "CodeReady Containers". Just this match/record is not performed now. It seems Zettlr has the same issue and also no known solution as of yet.

File: main.js [hunk 2]
```diff
@@ -186,6 +186,37 @@ createTrayMenu = function(state) {
...
 }
 
+const showNotification = function(title, body) {
+  new Notification({ title: title, body: body }).show()
+}
+
+const handleStartClick = async function() {
+    const startResult = await commander.start();
+    if (startResult.KubeletStarted) {
+      showNotification("Cluster started", "CodeReady Containers cluster has successfully started");
```
Comment (gbraad) [inline main.js]: 'has' => 'has been' or no 'has' at all?

File: main.js [hunk 3]
```diff
@@ -186,6 +186,37 @@ createTrayMenu = function(state) {
...
+    } else {
+      showNotification("Failed to start cluster", startResult.error);
+    }
+}
+
+const handleDeleteClick = async function() {
+  const deleteResult = await commander.delete();
+  if (deleteResult.deleted) {
+    showNotification("Cluster Deleted", "CodeReady Containers cluster is successfully deleted");
```
Comment (gbraad) [inline main.js]: "is" => ditto, don't use a different form or verb

File: main.js [hunk 4]
```diff
@@ -186,6 +186,37 @@ createTrayMenu = function(state) {
...
+  } else {
+    showNotification("Cluster not deleted", deleteResult.error);
+  }
+}
+
+const handleStopClick = async function() {
+  const stopResult = await commander.stop();
+  if (stopResult.stopped) {
+    showNotification("Cluster Stopped", "CodeReady Containers cluster has stopped")
```
Comment (gbraad) [inline main.js]: @robin-owen what would you say. Does this sound OK?

**Comment by anjannath** [comment]
@gbraad I updated the sentences to what we currently have on the native tray

**Comment by gbraad** [comment]
Please separate commits according to what is added.
This PR also contains two very different things.

This makes it harder to identify issues and do a review.

**Comment by anjannath** [comment]
> 
> 
> Please separate commits according to what is added. This PR also contains two very different things.
> 
> This makes it harder to identify issues and do a review.

The second commit about AppID is a small enough change and is related to notifications, as this is needed to show well formed toas notifications. so kept them in a single PR

Another part for displaying the notification is to add the same appID to the shortcut of the tray exe, which should be handled during installation

**Comment by gbraad** [comment]
@anjannath Does this need to be be added?

**Comment by anjannath** [comment]
@gbraad I need to have another look here, will fix merge conflicts and update

**Comment by anjannath** [comment]
Closing as most of the code is unusable now after the refactors and addition of react
Will file a new one

**Pull request closed without merging**